### ISSUE TITLE
Use of `static abstract` interface members on .NET 6 produces incorrect error

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn#65361]
**Version Used**: csc 4.5.0-2.22559.14 (5a8c698b)

**Steps to Reproduce**:

1. Create a C# library project that targets `net6.0` and specifies LangVersion=11.
2. Declare this interface:
```cs
public interface IComIID
{
    internal static abstract ref readonly Guid Guid { get; }
}
```

3. Build the project

**Expected Behavior**:

An error about using a .NET 7-only feature.

**Actual Behavior**:

An error about enabling preview features (that appears twice, once for each modifier):

> 1> error CA2252: Using both 'static' and 'abstract' modifiers requires opting into preview features. See https://aka.ms/dotnet-warnings/preview-features for more information.

In fact when I add EnablePreviewFeatures=true to the project file, the same error is produced, which looks quite broken.

**Comment by Youssef1313** [comment]
This belongs to dotnet/roslyn-analyzers

**Comment by Youssef1313** [comment]
Related discussion: https://github.com/dotnet/sdk/issues/28049

**Comment by AArnott** [comment]
I'm good with transferring this issue to one of the other repos. I just don't know which, and I lack permissions to do so.

**Comment by Youssef1313** [comment]
@mavasani Can you move to dotnet/roslyn-analyzers for now if you think CA2252 shouldn't complain about static abstracts being preview? The work on dotnet/sdk is already tracked by https://github.com/dotnet/sdk/issues/28049

**Comment by AlekseyTs** [comment]
>  I think the feature is no longer preview and it shouldn't require `EnablePreviewFeatures`? 

The feature is marked as preview in net6.0, certain APIs are specially marked in reference assemblies. Release of any other SDK isn't going to change that and usage of any other compiler version isn't going to change that. Therefore, I think `EnablePreviewFeatures` is still required as long as those reference assemblies are used. If I remember correctly, CA2252 is reported by an analyzer that is getting shipped with an SDK

**Comment by jcouv** [comment]
Closing as by-design (the APIs shipped in .NET 6 were preview and will remain so).

FWIW, the CA2252 [implementation](https://github.com/dotnet/roslyn-analyzers/blob/main/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs) is in the `roslyn-analyzers` repo.
